### PR TITLE
#2334 - fix create account after checkout bugs

### PR DESCRIPTION
**Pull request by tatiana-scandi**
Issue: https://github.com/scandipwa/scandipwa/issues/2443
Initial task: https://github.com/scandipwa/scandipwa/issues/2148

- Fixed issues 1, 3, 4 from #2443 
- Added label "You can create an account after checkout" to guest checkout page
- Had to remove a whole component (`CheckoutRegisterForm`) and rewrite logic to redirect to create account page for account creation, as the previous implementation didn't meet the requirements described in #2148

### CODE REVIEW

File: packages/scandipwa/src/component/MyAccountCreateAccount/MyAccountCreateAccount.component.js [hunk 1]
```diff
@@ -24,12 +25,7 @@ export class MyAccountCreateAccount extends PureComponent {
...
         handleSignIn: PropTypes.func.isRequired,
         showTaxVatNumber: PropTypes.string.isRequired,
-        vatNumberValidation: PropTypes.array.isRequired,
-        defaultValues: PropTypes.object
-    };
-
-    static defaultProps = {
-        defaultValues: {}
+        vatNumberValidation: PropTypes.array.isRequired
     };
```
Comment (alfredsgenkins) [inline packages/scandipwa/src/component/MyAccountCreateAccount/MyAccountCreateAccount.component.js:29]: Please, do not remove existing functinality for no reason. Ideally I would make the container update these fields if state is present, so no logic is left in component.

File: packages/scandipwa/src/route/Checkout/Checkout.component.js [hunk 2]
```diff
@@ -217,12 +220,17 @@ export class Checkout extends PureComponent {
     }
 
     renderDetailsStep() {
-        const { orderID, setLoading } = this.props;
+        const {
+            orderID, isEmailAvailable, email, billingAddress: { firstname, lastname }
```
Comment (alfredsgenkins) [inline packages/scandipwa/src/route/Checkout/Checkout.component.js:224]: Put each prop on new line fro better reading.

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]
Minor change needed. Also why did the `Create free account and keep track of your orders` translation got removed? It is indeed unused?

File: packages/scandipwa/src/component/MyAccountCreateAccount/MyAccountCreateAccount.component.js (hunk 1 above)
Comment (tatiana-scandi) [reply packages/scandipwa/src/component/MyAccountCreateAccount/MyAccountCreateAccount.component.js:29]: The reason for removal of default props is that the component doesn't use `defaultValues` anymore. This line has also been removed:
```const { defaultValues: { email = '' } } = this.props;```

And props `defaultValues` is not passed to `MyAccountCreateAccount` by any other component any more.

**Comment by tatiana-scandi** [comment]
> Also why did the `Create free account and keep track of your orders` translation got removed? It is indeed unused?

Yes, it has been removed in #2148, but the translation was not.

You see a warning in the console, if there are unused translations, it is hard to pass by. By the way, is there a rule/setting to show a warning in the opposite case - when you use a translatable string, which is missing in the translation files?

**Comment by alfredsgenkins** [comment]
It automatically ads it, and I think prints out something about it, not sure, not sure. Ping @yeegor for this precise instruction.

**Comment by yeegor** [comment]
It should print a message on the initial compilation. The appended translation will be visible in all the localisation files.

**Pull request merged**